garnish-modules/garnish-modules-generic/garnish-httpclient/src/main/java/org/garnishtest/modules/generic/httpclient/RestApiClientFactory.java

```diff
@@ -57,75 +53,29 @@ public void setMaxConnectionsPerRoute(final int maxConnectionsPerRoute) {
     }
 
     public void setConnectionTimeoutMillis(final int connectionTimeoutMillis) {
-        this.connectionTimeoutMillis = connectionTimeoutMillis;
+        this.connectionTimeoutMillis = Duration.ofMillis(connectionTimeoutMillis);
     }
 
     public void setSocketTimeoutMillis(final int socketTimeoutMillis) {
-        this.socketTimeoutMillis = socketTimeoutMillis;
+        this.socketTimeoutMillis = Duration.ofMillis(socketTimeoutMillis);
     }
 
-    @NonNull
-    public HttpClient createClient() {
-        final HttpClientConnectionManager connectionManager = createHttpClientConnectionManager(
-                this.maxTotalConnections,
-                this.maxConnectionsPerRoute
-        );
-
-        final ConnectionConfig connectionConfig = getConnectionConfig();
-        final HttpRoutePlanner httpRoutePlanner = getHttpRoutePlanner();
-        final RequestConfig requestConfig = getRequestConfig(this.connectionTimeoutMillis, this.socketTimeoutMillis);
-
-        return HttpClients.custom()
-                          .setConnectionManager(connectionManager)
-                          .setDefaultConnectionConfig(connectionConfig)
-                          .setRoutePlanner(httpRoutePlanner)
-                          .setDefaultRequestConfig(requestConfig)
-                          .disableRedirectHandling()
-                          .build();
+    @NonNull public HttpClient createClient() {
+        System.setProperty("jdk.httpclient.connectionPoolSize",
```

Comment:
By default, a connection pool is created if you build a standard HttpClient. Each HttpClient has a single implementation-specific thread that polls all of its connections.
If we want to change the keep-alive timeout or connection pool size, we can do so using the System property.
However the value is only read once when the class jdk.internal.net.http.ConnectionPool is loaded. Afterward, it can't be changed anymore.